harvester/config.go

```diff
@@ -0,0 +1,242 @@
+package harvester
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/ghodss/yaml"
+)
+
+func UnmarshalDiskInfo(data []byte) (DiskInfo, error) {
+	var r DiskInfo
+	err := json.Unmarshal(data, &r)
+	return r, err
+}
+
+func (r *DiskInfo) Marshal() ([]byte, error) {
+	return json.Marshal(r)
+}
+
+type DiskInfo struct {
+	Disks []Disk `json:"disks"`
+}
+
+type Disk struct {
+	ImageName        string `json:"imageName"`
+	StorageClassName string `json:"storageClassName"`
+
+	Size      int `json:"size"`
+	BootOrder int `json:"bootOrder"`
+
+	Bus  string `json:"bus"`
+	Type string `json:"type"`
+
+	HotPlugAble bool `json:"hotPlugAble"`
+}
+
+func UnmarshalNetworkInfo(data []byte) (NetworkInfo, error) {
+	var r NetworkInfo
+	err := json.Unmarshal(data, &r)
+	return r, err
+}
+
+func (r *NetworkInfo) Marshal() ([]byte, error) {
+	return json.Marshal(r)
+}
+
+type NetworkInfo struct {
+	NetworkInterfaces []NetworkInterface `json:"interfaces"`
+}
+
+type NetworkInterface struct {
+	NetworkName string `json:"networkName"`
```

Comment:
cc @yaocw2020  please take a look of this definition, can it cover all potential networks ? thanks.

Reply:
I believe it's enough currently.